Druid driver with auth fails sync

### Describe the bug

I've added a druid database, using _Authentication header_ - providing basic auth credentials. This immediately spewed an error, but... the database has been added and the datasource has been synced. Later when I visit metabase admin panel and try to sync this database, the request to sync will always fail.

### To Reproduce

1. Go to Databases in Admin panel
2. Create DB that uses Druid engine/driver
3. Use Authentication header to authenticate with basic auth
4. See error in logs/console


### Expected behavior

Properly authenticate requests when Druid uses basic auth.

### Logs
```
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.task.sync-databases Scheduling sync/analyze for database 22: trigger: metabase.task.sync-and-analyze.trigger.22
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.task.sync-databases Scheduling field-values for database 22: trigger: metabase.task.update-field-values.trigger.22
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: Sync druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: Sync metadata for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 ERROR metabase.server.middleware.log POST /api/database 500 175.8 ms (23 DB calls) 
{:via
 [{:type clojure.lang.ExceptionInfo,
   :message "clj-http: status 401",
   :data
   {:type :db,
    :request-url "https://druid-test-router.DOMAIN:443/status",
    :request-options {:content-type "application/json;charset=UTF-8"},
    :response {:error "Unauthorized request", :errorMessage nil, :errorClass nil, :host "druid-router"}},
   :at [metabase.driver.druid.client$do_request invokeStatic "client.clj" 49]}
  {:type clojure.lang.ExceptionInfo,
   :message "clj-http: status 401",
   :data
   {:cached nil,
    :request-time 9,
    :repeatable? false,
    :protocol-version {:name "HTTP", :major 1, :minor 1},
    :streaming? true,
    :http-client
    #object[org.apache.http.impl.client.InternalHttpClient 0x2e11e5a0 "org.apache.http.impl.client.InternalHttpClient@2e11e5a0"],
    :chunked? false,
    :type :clj-http.client/unexceptional-status,
    :reason-phrase "Unauthorized",
    :headers
    {"Date" "Thu, 18 Apr 2024 14:44:29 GMT",
     "Content-Type" "application/json;charset=utf-8",
     "Content-Length" "92",
     "Connection" "close",
     "Cache-Control" "no-cache, no-store, max-age=0",
     "Content-Security-Policy" "frame-ancestors 'none'",
     "WWW-Authenticate" "Basic"},
    :orig-content-encoding nil,
    :status 401,
    :length 92,
    :body "{\"error\":\"Unauthorized request\",\"errorMessage\":null,\"errorClass\":null,\"host\":\"druid-router\"}",
    :trace-redirects []},
   :at [slingshot.support$stack_trace invoke "support.clj" 201]}],
 :trace
 [[slingshot.support$stack_trace invoke "support.clj" 201]
  [clj_http.client$exceptions_response invokeStatic "client.clj" 241]
  [clj_http.client$exceptions_response invoke "client.clj" 232]
  [clj_http.client$wrap_exceptions$fn__27728 invoke "client.clj" 250]
  [clj_http.client$wrap_accept$fn__27940 invoke "client.clj" 728]
  [clj_http.client$wrap_accept_encoding$fn__27947 invoke "client.clj" 750]
  [clj_http.client$wrap_content_type$fn__27934 invoke "client.clj" 711]
  [clj_http.client$wrap_form_params$fn__28045 invoke "client.clj" 959]
  [clj_http.client$wrap_nested_params$fn__28066 invoke "client.clj" 993]
  [clj_http.client$wrap_flatten_nested_params$fn__28075 invoke "client.clj" 1017]
  [clj_http.client$wrap_method$fn__28013 invoke "client.clj" 893]
  [clj_http.cookies$wrap_cookies$fn__26977 invoke "cookies.clj" 128]
  [clj_http.links$wrap_links$fn__27405 invoke "links.clj" 63]
  [clj_http.client$wrap_unknown_host$fn__28083 invoke "client.clj" 1046]
  [clj_http.client$request_STAR_ invokeStatic "client.clj" 1174]
  [clj_http.client$request_STAR_ invoke "client.clj" 1167]
  [clj_http.client$get invokeStatic "client.clj" 1180]
  [clj_http.client$get doInvoke "client.clj" 1176]
  [clojure.lang.RestFn invoke "RestFn.java" 423]
  [metabase.driver.druid.client$do_request invokeStatic "client.clj" 35]
  [metabase.driver.druid.client$do_request doInvoke "client.clj" 22]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [clojure.core$partial$fn__5908 invoke "core.clj" 2641]
  [metabase.driver.druid.sync$dbms_version$fn__114651 invoke "sync.clj" 68]
  [metabase.util.ssh$do_with_ssh_tunnel invokeStatic "ssh.clj" 165]
  [metabase.util.ssh$do_with_ssh_tunnel invoke "ssh.clj" 154]
  [metabase.driver.druid.sync$dbms_version invokeStatic "sync.clj" 67]
  [metabase.driver.druid.sync$dbms_version invoke "sync.clj" 63]
  [metabase.driver.druid$fn__114749 invokeStatic "druid.clj" 34]
  [metabase.driver.druid$fn__114749 invoke "druid.clj" 32]
  [clojure.lang.MultiFn invoke "MultiFn.java" 234]
  [metabase.api.database$fn__98471 invokeStatic "database.clj" 809]
  [metabase.api.database$fn__98471 invoke "database.clj" 764]
  [compojure.core$wrap_response$fn__44649 invoke "core.clj" 160]
  [compojure.core$wrap_route_middleware$fn__44633 invoke "core.clj" 132]
  [compojure.core$wrap_route_info$fn__44638 invoke "core.clj" 139]
  [compojure.core$wrap_route_matches$fn__44642 invoke "core.clj" 151]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__44642 invoke "core.clj" 152]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661 invoke "core.clj" 200]
  [metabase.server.middleware.auth$enforce_authentication$fn__93927 invoke "auth.clj" 17]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661 invoke "core.clj" 200]
  [compojure.core$make_context$handler__44689 invoke "core.clj" 290]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 300]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 301]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [metabase.api.routes$fn__101920$fn__101923 invoke "routes.clj" 67]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661 invoke "core.clj" 200]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 667]
  [clojure.core$apply invoke "core.clj" 662]
  [metabase.server.routes$fn__102085$fn__102086 doInvoke "routes.clj" 72]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661 invoke "core.clj" 200]
  [compojure.core$make_context$handler__44689 invoke "core.clj" 290]
  [compojure.core$make_context$fn__44693 invoke "core.clj" 300]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__44642 invoke "core.clj" 153]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__44642 invoke "core.clj" 153]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__44642 invoke "core.clj" 153]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661$f__44662$respond_SINGLEQUOTE___44663 invoke "core.clj" 197]
  [metabase.server.routes$fn__102070$fn__102072 invoke "routes.clj" 49]
  [compojure.core$routes$fn__44661$f__44662 invoke "core.clj" 198]
  [compojure.core$routes$fn__44661 invoke "core.clj" 200]
  [metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__98693 invoke "exceptions.clj" 108]
  [metabase.server.middleware.exceptions$catch_api_exceptions$fn__98690 invoke "exceptions.clj" 96]
  [metabase.server.middleware.log$log_api_call$fn__102353$fn__102354$fn__102355 invoke "log.clj" 216]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invokeStatic "diagnostic.clj" 18]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invoke "diagnostic.clj" 12]
  [metabase.server.middleware.log$log_api_call$fn__102353$fn__102354 invoke "log.clj" 208]
  [toucan2.execute$do_with_call_counts invokeStatic "execute.clj" 112]
  [toucan2.execute$do_with_call_counts invoke "execute.clj" 103]
  [metabase.server.middleware.log$log_api_call$fn__102353 invoke "log.clj" 207]
  [metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__104389 invoke "browser_cookie.clj" 40]
  [metabase.server.middleware.security$add_security_headers$fn__84814 invoke "security.clj" 182]
  [metabase.server.middleware.json$wrap_json_body$fn__46006 invoke "json.clj" 67]
  [metabase.server.middleware.offset_paging$handle_paging$fn__84838 invoke "offset_paging.clj" 45]
  [metabase.server.middleware.json$wrap_streamed_json_response$fn__46024 invoke "json.clj" 103]
  [ring.middleware.keyword_params$wrap_keyword_params$fn__104656 invoke "keyword_params.clj" 55]
  [ring.middleware.params$wrap_params$fn__104675 invoke "params.clj" 77]
  [metabase.server.middleware.misc$maybe_set_site_url$fn__67134 invoke "misc.clj" 61]
  [metabase.server.middleware.session$reset_session_timeout$fn__72557 invoke "session.clj" 543]
  [metabase.server.middleware.session$bind_current_user$fn__72523$fn__72524 invoke "session.clj" 438]
  [metabase.server.middleware.session$do_with_current_user invokeStatic "session.clj" 417]
  [metabase.server.middleware.session$do_with_current_user invoke "session.clj" 401]
  [metabase.server.middleware.session$bind_current_user$fn__72523 invoke "session.clj" 437]
  [metabase.server.middleware.session$wrap_current_user_info$fn__72506 invoke "session.clj" 376]
  [metabase.server.middleware.session$wrap_session_id$fn__72478 invoke "session.clj" 255]
  [metabase.server.middleware.auth$wrap_static_api_key$fn__93935 invoke "auth.clj" 30]
  [ring.middleware.cookies$wrap_cookies$fn__104576 invoke "cookies.clj" 194]
  [metabase.server.middleware.misc$add_content_type$fn__67116 invoke "misc.clj" 29]
  [metabase.server.middleware.misc$disable_streaming_buffering$fn__67142 invoke "misc.clj" 78]
  [ring.middleware.gzip$wrap_gzip$fn__104618 invoke "gzip.clj" 86]
  [metabase.server.middleware.misc$bind_request$fn__67145 invoke "misc.clj" 95]
  [metabase.server.middleware.ssl$redirect_to_https_middleware$fn__104405 invoke "ssl.clj" 41]
  [metabase.server$async_proxy_handler$fn__67556 invoke "server.clj" 78]
  [metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a handle nil -1]
  [org.eclipse.jetty.server.handler.StatisticsHandler handle "StatisticsHandler.java" 173]
  [org.eclipse.jetty.server.handler.HandlerWrapper handle "HandlerWrapper.java" 122]
  [org.eclipse.jetty.server.Server handle "Server.java" 563]
  [org.eclipse.jetty.server.HttpChannel$RequestDispatchable dispatch "HttpChannel.java" 1598]
  [org.eclipse.jetty.server.HttpChannel dispatch "HttpChannel.java" 753]
  [org.eclipse.jetty.server.HttpChannel handle "HttpChannel.java" 501]
  [org.eclipse.jetty.server.HttpConnection onFillable "HttpConnection.java" 287]
  [org.eclipse.jetty.io.AbstractConnection$ReadCallback succeeded "AbstractConnection.java" 314]
  [org.eclipse.jetty.io.FillInterest fillable "FillInterest.java" 100]
  [org.eclipse.jetty.io.SelectableChannelEndPoint$1 run "SelectableChannelEndPoint.java" 53]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy runTask "AdaptiveExecutionStrategy.java" 421]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy consumeTask "AdaptiveExecutionStrategy.java" 390]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy tryProduce "AdaptiveExecutionStrategy.java" 277]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy run "AdaptiveExecutionStrategy.java" 199]
  [org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread run "ReservedThreadExecutor.java" 411]
  [org.eclipse.jetty.util.thread.QueuedThreadPool runJob "QueuedThreadPool.java" 969]
  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner doRunJob "QueuedThreadPool.java" 1194]
  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner run "QueuedThreadPool.java" 1149]
  [java.lang.Thread run nil -1]],
 :cause "clj-http: status 401",
 :data
 {:cached nil,
  :request-time 9,
  :repeatable? false,
  :protocol-version {:name "HTTP", :major 1, :minor 1},
  :streaming? true,
  :http-client
  #object[org.apache.http.impl.client.InternalHttpClient 0x2e11e5a0 "org.apache.http.impl.client.InternalHttpClient@2e11e5a0"],
  :chunked? false,
  :type :clj-http.client/unexceptional-status,
  :reason-phrase "Unauthorized",
  :headers
  {"Date" "Thu, 18 Apr 2024 14:44:29 GMT",
   "Content-Type" "application/json;charset=utf-8",
   "Content-Length" "92",
   "Connection" "close",
   "Cache-Control" "no-cache, no-store, max-age=0",
   "Content-Security-Policy" "frame-ancestors 'none'",
   "WWW-Authenticate" "Basic"},
  :orig-content-encoding nil,
  :status 401,
  :length 92,
  :body "{\"error\":\"Unauthorized request\",\"errorMessage\":null,\"errorClass\":null,\"host\":\"druid-router\"}",
  :trace-redirects []},
 :message "clj-http: status 401",
 :request-url "https://druid-test-router.DOMAIN:443/status",
 :request-options {:content-type "application/json;charset=UTF-8"},
 :response {:error "Unauthorized request", :errorMessage nil, :errorClass nil, :host "druid-router"}}

[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-dbms-version'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 WARN metabase.sync.util Error running step ''sync-dbms-version'' for druid Database 22 ''druid-test2'',clojure.lang.ExceptionInfo: clj-http: status 401 {:type :db, :request-url "https://druid-test-router.DOMAIN:443/status", :request-options {:content-type "application/json;charset=UTF-8"}, :response {:error "Unauthorized request", :errorMessage nil, :errorClass nil, :host "druid-router"}},	at metabase.driver.druid.client$do_request.invokeStatic(client.clj:49),	at metabase.driver.druid.client$do_request.doInvoke(client.clj:22),	at clojure.lang.RestFn.invoke(RestFn.java:425),	at clojure.core$partial$fn__5908.invoke(core.clj:2641),	at metabase.driver.druid.sync$dbms_version$fn__114651.invoke(sync.clj:68),	at metabase.util.ssh$do_with_ssh_tunnel.invokeStatic(ssh.clj:165),	at metabase.util.ssh$do_with_ssh_tunnel.invoke(ssh.clj:154),	at metabase.driver.druid.sync$dbms_version.invokeStatic(sync.clj:67),	at metabase.driver.druid.sync$dbms_version.invoke(sync.clj:63),	at metabase.driver.druid$fn__114749.invokeStatic(druid.clj:34),	at metabase.driver.druid$fn__114749.invoke(druid.clj:32),	at clojure.lang.MultiFn.invoke(MultiFn.java:234),	at metabase.sync.sync_metadata.dbms_version$sync_dbms_version_BANG_.invokeStatic(dbms_version.clj:20),	at metabase.sync.sync_metadata.dbms_version$sync_dbms_version_BANG_.invoke(dbms_version.clj:16),	at clojure.lang.AFn.applyToHelper(AFn.java:154),	at clojure.lang.AFn.applyTo(AFn.java:144),	at clojure.core$apply.invokeStatic(core.clj:669),	at clojure.core$apply.invoke(core.clj:662),	at metabase.sync.util$run_step_with_metadata$fn__60304.doInvoke(util.clj:462),	at clojure.lang.RestFn.invoke(RestFn.java:397),	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:131),	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:125),	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:149),	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:145),	at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:457),	at metabase.sync.util$run_step_with_metadata.invoke(util.clj:452),	at metabase.sync.util$run_sync_operation$fn__60383.invoke(util.clj:568),	at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:566),	at metabase.sync.util$run_sync_operation.invoke(util.clj:560),	at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__82369.invoke(sync_metadata.clj:70),	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:190),	at metabase.sync.util$do_with_error_handling.invoke(util.clj:183),	at clojure.core$partial$fn__5910.invoke(core.clj:2647),	at metabase.driver$fn__49798.invokeStatic(driver.clj:778),	at metabase.driver$fn__49798.invoke(driver.clj:778),	at clojure.lang.MultiFn.invoke(MultiFn.java:239),	at metabase.sync.util$sync_in_context$fn__60218.invoke(util.clj:166),	at metabase.sync.util$with_db_logging_disabled$fn__60215.invoke(util.clj:158),	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:131),	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:125),	at metabase.sync.util$with_start_and_finish_logging$fn__60202.invoke(util.clj:143),	at metabase.sync.util$with_sync_events$fn__60197.invoke(util.clj:117),	at metabase.sync.util$with_duplicate_ops_prevented$fn__60184.invoke(util.clj:89),	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:215),	at metabase.sync.util$do_sync_operation.invoke(util.clj:209),	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:68),	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:65),	at metabase.sync$sync_database_BANG_$fn__82379.invoke(sync.clj:52),	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:190),	at metabase.sync.util$do_with_error_handling.invoke(util.clj:183),	at clojure.core$partial$fn__5910.invoke(core.clj:2647),	at metabase.driver$fn__49798.invokeStatic(driver.clj:778),	at metabase.driver$fn__49798.invoke(driver.clj:778),	at clojure.lang.MultiFn.invoke(MultiFn.java:239),	at metabase.sync.util$sync_in_context$fn__60218.invoke(util.clj:166),	at metabase.sync.util$with_db_logging_disabled$fn__60215.invoke(util.clj:158),	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:131),	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:125),	at metabase.sync.util$with_start_and_finish_logging$fn__60202.invoke(util.clj:143),	at metabase.sync.util$with_sync_events$fn__60197.invoke(util.clj:117),	at metabase.sync.util$with_duplicate_ops_prevented$fn__60184.invoke(util.clj:89),	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:215),	at metabase.sync.util$do_sync_operation.invoke(util.clj:209),	at metabase.sync$sync_database_BANG_.invokeStatic(sync.clj:51),	at metabase.sync$sync_database_BANG_.invoke(sync.clj:37),	at metabase.sync$sync_database_BANG_.invokeStatic(sync.clj:46),	at metabase.sync$sync_database_BANG_.invoke(sync.clj:37),	at metabase.events.sync_database$publish_event_BANG__primary_method_metabase_events_sync_database_event$fn__104093.invoke(sync_database.clj:24),	at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047),	at clojure.lang.AFn.call(AFn.java:18),	at java.base/java.util.concurrent.FutureTask.run(Unknown Source),	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source),	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source),	at java.base/java.lang.Thread.run(Unknown Source),Caused by: clojure.lang.ExceptionInfo: clj-http: status 401 {:cached nil, :request-time 8, :repeatable? false, :protocol-version {:name "HTTP", :major 1, :minor 1}, :streaming? true, :http-client #object[org.apache.http.impl.client.InternalHttpClient 0x55c644dc "org.apache.http.impl.client.InternalHttpClient@55c644dc"], :chunked? false, :type :clj-http.client/unexceptional-status, :reason-phrase "Unauthorized", :headers {"Date" "Thu, 18 Apr 2024 14:44:29 GMT", "Content-Type" "application/json;charset=utf-8", "Content-Length" "92", "Connection" "close", "Cache-Control" "no-cache, no-store, max-age=0", "Content-Security-Policy" "frame-ancestors 'none'", "WWW-Authenticate" "Basic"}, :orig-content-encoding nil, :status 401, :length 92, :body "{\"error\":\"Unauthorized request\",\"errorMessage\":null,\"errorClass\":null,\"host\":\"druid-router\"}", :trace-redirects []},	at slingshot.support$stack_trace.invoke(support.clj:201),	at clj_http.client$exceptions_response.invokeStatic(client.clj:241),	at clj_http.client$exceptions_response.invoke(client.clj:232),	at clj_http.client$wrap_exceptions$fn__27728.invoke(client.clj:250),	at clj_http.client$wrap_accept$fn__27940.invoke(client.clj:728),	at clj_http.client$wrap_accept_encoding$fn__27947.invoke(client.clj:750),	at clj_http.client$wrap_content_type$fn__27934.invoke(client.clj:711),	at clj_http.client$wrap_form_params$fn__28045.invoke(client.clj:959),	at clj_http.client$wrap_nested_params$fn__28066.invoke(client.clj:993),	at clj_http.client$wrap_flatten_nested_params$fn__28075.invoke(client.clj:1017),	at clj_http.client$wrap_method$fn__28013.invoke(client.clj:893),	at clj_http.cookies$wrap_cookies$fn__26977.invoke(cookies.clj:128),	at clj_http.links$wrap_links$fn__27405.invoke(links.clj:63),	at clj_http.client$wrap_unknown_host$fn__28083.invoke(client.clj:1046),	at clj_http.client$request_STAR_.invokeStatic(client.clj:1174),	at clj_http.client$request_STAR_.invoke(client.clj:1167),	at clj_http.client$get.invokeStatic(client.clj:1180),	at clj_http.client$get.doInvoke(client.clj:1176),	at clojure.lang.RestFn.invoke(RestFn.java:423),	at metabase.driver.druid.client$do_request.invokeStatic(client.clj:35),	... 73 more
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-dbms-version'' for druid Database 22 ''druid-test2'' (10.2 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-timezone'' for druid Database 22 ''druid-test2'' (145.6 µs)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-tables'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-timezone'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.sync-metadata.sync-timezone :druid database 22 default timezone is nil
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.sync-metadata.tables Found new tables: (Table  ''wikipedia'')
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.sync-metadata.tables Updating table metadata for Table 4659 ''wikipedia''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 DEBUG metabase.server.middleware.log GET /api/database 200 6.4 ms (3 DB calls) App DB connections: 2/14 Jetty threads: 3/50 (4 idle, 0 queued) (126 total active threads) Queries in flight: 0 (0 queued)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-tables'' for druid Database 22 ''druid-test2'' (112.0 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-fields'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-fields'' for druid Database 22 ''druid-test2'' (171.9 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-fks'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-fks'' for druid Database 22 ''druid-test2'' (6.4 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-indexes'' for druid Database 22 ''druid-test2'' (29.1 µs)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-metabase-metadata'' for druid Database 22 ''druid-test2'' (43.8 µs)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: step ''sync-table-privileges'' for druid Database 22 ''druid-test2'' (29.8 µs)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-indexes'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-metabase-metadata'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''sync-table-privileges'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util FINISHED: Sync metadata for druid Database 22 ''druid-test2'' (403.8 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: Analyze data for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.util STARTING: step ''fingerprint-fields'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 INFO metabase.sync.analyze fingerprint-fields Analyzed [************······································] 😒   25% Table 4659 ''wikipedia''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:29+02:00 WARN metabase.driver.druid.query-processor scan queries can only be sorted by the 'timestamp' column.
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:31+02:00 DEBUG metabase.server.middleware.log GET /api/database 200 7.0 ms (3 DB calls) App DB connections: 1/14 Jetty threads: 3/50 (4 idle, 0 queued) (127 total active threads) Queries in flight: 0 (0 queued)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util FINISHED: step ''fingerprint-fields'' for druid Database 22 ''druid-test2'' (4.1 s)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util STARTING: step ''classify-fields'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.analyze classify-fields Analyzed [*************************·························] 😬   50% Table 4659 ''wikipedia''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util FINISHED: step ''classify-fields'' for druid Database 22 ''druid-test2'' (110.9 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util STARTING: step ''classify-tables'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.analyze classify-tables Analyzed [*************************************·············] 😋   75% Table 4659 ''wikipedia''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util FINISHED: step ''classify-tables'' for druid Database 22 ''druid-test2'' (6.4 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util FINISHED: Analyze data for druid Database 22 ''druid-test2'' (4.3 s)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util STARTING: Cache field values in druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util STARTING: step ''delete-expired-advanced-field-values'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util FINISHED: step ''delete-expired-advanced-field-values'' for druid Database 22 ''druid-test2'' (76.1 ms)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:33+02:00 INFO metabase.sync.util STARTING: step ''update-field-values'' for druid Database 22 ''druid-test2''
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:35+02:00 INFO metabase.sync.util FINISHED: step ''update-field-values'' for druid Database 22 ''druid-test2'' (2.0 s)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:35+02:00 INFO metabase.sync.util FINISHED: Cache field values in druid Database 22 ''druid-test2'' (2.1 s)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:35+02:00 INFO metabase.sync.util FINISHED: Sync druid Database 22 ''druid-test2'' (6.8 s)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:43+02:00 DEBUG metabase.server.middleware.log GET /api/user/current 200 12.5 ms (10 DB calls) App DB connections: 2/14 Jetty threads: 4/50 (3 idle, 0 queued) (127 total active threads) Queries in flight: 0 (0 queued)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:43+02:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 14.1 ms (7 DB calls) App DB connections: 1/14 Jetty threads: 3/50 (4 idle, 0 queued) (127 total active threads) Queries in flight: 0 (0 queued)
[53f4eed0-9bd8-4c1f-a417-2b3c4636099a] 2024-04-18T16:44:43+02:00 DEBUG metabase.server.middleware.log GET /api/database 200 6.7 ms (3 DB calls) App DB connections: 1/14 Jetty threads: 4/50 (3 idle, 0 queued) (127 total active threads) Queries in flight: 0 (0 queued)
```

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Linux",
    "os.version": "5.10.184-175.731.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "athena",
      "druid",
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.10"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-04-02",
      "tag": "v0.49.3",
      "hash": "dba0992"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```


### Severity

High

### Additional context

PR introducing auth to druid driver: https://github.com/metabase/metabase/pull/20412

--- a/modules/drivers/druid/src/metabase/driver/druid/sync.clj
+++ b/modules/drivers/druid/src/metabase/driver/druid/sync.clj
@@ -66,5 +66,10 @@
   [database]
   {:pre [(map? (:details database))]}
   (ssh/with-ssh-tunnel [details-with-tunnel (:details database)]
-    (-> (druid.client/GET (druid.client/details->url details-with-tunnel "/status"))
+    (-> (druid.client/GET (druid.client/details->url details-with-tunnel "/status")
+          :auth-enabled     (-> database :details :auth-enabled)
+          :auth-username    (-> database :details :auth-username)
+          :auth-token-value (-> (:details database)
+                                (secret/db-details-prop->secret-map "auth-token")
+                                secret/value->string))
         (select-keys [:version]))))

--- a/modules/drivers/druid/test/metabase/driver/druid/sync_test.clj
+++ b/modules/drivers/druid/test/metabase/driver/druid/sync_test.clj
@@ -4,7 +4,10 @@
    [malli.core :as mc]
    [malli.error :as me]
    [metabase.driver :as driver]
+   [metabase.driver.druid.client :as druid.client]
+   [metabase.driver.druid.sync :as druid.sync]
    [metabase.models.database :refer [Database]]
+   [metabase.models.secret :as secret]
    [metabase.sync.sync-metadata.dbms-version :as sync-dbms-ver]
    [metabase.test :as mt]
    [metabase.timeseries-query-processor-test.util :as tqpt]
@@ -60,3 +63,22 @@
             (is (nil? version-after-update)))
           (testing "Check that the value was set again after sync"
             (is (nil? (check-dbms-version (db-dbms-version db))))))))))
+
+(deftest ^:synchronized auth-dbms-version-test
+  (mt/test-driver
+   :druid
+   (testing "`dbms-version` uses auth parameters (#41579)"
+     (tqpt/with-flattened-dbdef
+       (let [get-args (volatile! nil)
+             db-with-auth-details (update (mt/db) :details assoc
+                                          :auth-enabled     true
+                                          :auth-username    "admin"
+                                          :auth-token-value "password1")]
+         ;; Following ensures that auth parameters are passed to GET during version sync if present.
+         (with-redefs [secret/value->string (constantly "password1")
+                       druid.client/GET (fn [_url & params] (vreset! get-args (apply hash-map params)) nil)]
+           ;; Just fill in the params with internally modified `dbms-version`.
+           (druid.sync/dbms-version db-with-auth-details)
+           (is (= true        (:auth-enabled     @get-args)))
+           (is (= "admin"     (:auth-username    @get-args)))
+           (is (= "password1" (:auth-token-value @get-args)))))))))

